### ISSUE TITLE
Optimizer with attributes and direct mode not possible

### ISSUE THREAD

**Issue by jd-lara** [odow/SDDP.jl#382]
A specification like this fails. 

```Julia
    model = SDDP.MarkovianPolicyGraph(
        transition_matrices = problem.ext["MARKOV_TRANSITION"],
        sense = :Min,
        lower_bound = 0.0,
        optimizer =  JuMP.optimizer_with_attributes(CPLEX.Optimizer) #not pasting all the params to keep it short,
        direct_mode = true
    ) do sp, node
```

```Julia
│     [2] SDDP.PolicyGraph(::var"#77#94"{UnitRange{Int64},Array{String,1},Array{String,1},Array{String,1},Array{String,1},Dict{String,VariableCost{Array{Tuple{Float64,Float64},1}}},Dict{String,NamedTuple{(:min, :max),Tuple{Float64,Float64}}},Dict{String,Float64},Array{Float64,1},Array{Float64,1},Array{Float64,1},Array{Float64,1},Array{Float64,2},Dict{String,Bool},Dict{String,Float64},Dict{String,Float64},Dict{String,Float64},Dict{String,Float64},Float64}, ::SDDP.Graph{Tuple{Int64,Int64}}; sense::Symbol, lower_bound::Float64, upper_bound::Float64, optimizer::MathOptInterface.OptimizerWithAttributes, bellman_function::Nothing, direct_mode::Bool, integrality_handler::SDDP.ContinuousRelaxation) at /Users/jdlara/.julia/packages/SDDP/f1WSH/src/user_interface.jl:612
│     [3] #MarkovianPolicyGraph#12 at /Users/jdlara/.julia/packages/SDDP/f1WSH/src/user_interface.jl:530 [inlined]
```

**Comment by odow** [comment]
I need to use `MOI.instantiate` here: 
https://github.com/odow/SDDP.jl/blob/c6951f15b3607e88789d211c264dbea5ae141dcf/src/user_interface.jl#L473

**Comment by odow** [comment]
See https://github.com/odow/SDDP.jl/pull/383.

As a work-around, you could also set the parameters in the model with `set_optimizer_attribute(sp, ...)`, or pass `() -> MOI.instantiate(optimizer_with_attributes(CPLEX.Optimizer))`

**Comment by odow** [comment]
Closed by #383
